modules/market/src/test/java/com/opengamma/strata/market/interpolator/InterpolatorCurveExtrapolatorTest.java

```diff
@@ -90,49 +83,42 @@ public void differentIntervalsTest() {
-      leftKeys[i] = xValues.get(0) - interval * i;
-      rightKeys[i] = xValues.get(nData - 1) + interval * i;
+      keys[i] = xValues.get(0) + interval * i;
     }
 
     CurveExtrapolator extrap = InterpolatorCurveExtrapolator.INSTANCE;
     int yDim = yValues.length;
     for (int k = 0; k < yDim; ++k) {
       BoundCurveInterpolator boundInterp =
-          CurveInterpolators.LOG_NATURAL_CUBIC_MONOTONE.bind(xValues, yValues[k], extrap, extrap);
-      Interpolator1D baseInterp = ((StandardBoundCurveInterpolator) boundInterp).getInterpolator();
+          CurveInterpolators.SQUARE_LINEAR.bind(xValues, yValues[k], extrap, extrap);
+      Interpolator1D baseInterp = new SquareLinearInterpolator1D();
       Interpolator1DDataBundle baseBundle =
           baseInterp.getDataBundleFromSortedArrays(xValues.toArray(), yValues[k].toArray());
       for (int j = 0; j < nKeys; ++j) {
         // value
-        assertEquals(boundInterp.interpolate(leftKeys[j]), baseInterp.interpolate(baseBundle, leftKeys[j]), TOL);
-        assertEquals(boundInterp.interpolate(rightKeys[j]), baseInterp.interpolate(baseBundle, rightKeys[j]), TOL);
+        assertEquals(boundInterp.interpolate(keys[j]), baseInterp.interpolate(baseBundle, keys[j]), TOL);
         // derivative 
-        assertEquals(boundInterp.firstDerivative(leftKeys[j]), baseInterp.firstDerivative(baseBundle, leftKeys[j]), TOL);
-        assertEquals(boundInterp.firstDerivative(rightKeys[j]), baseInterp.firstDerivative(baseBundle, rightKeys[j]), TOL);
+        assertEquals(boundInterp.firstDerivative(keys[j]), baseInterp.firstDerivative(baseBundle, keys[j]), TOL);
         // sensitivity
-        assertTrue(DoubleArrayMath.fuzzyEquals(boundInterp.parameterSensitivity(leftKeys[j]).toArray(),
-            baseInterp.getNodeSensitivitiesForValue(baseBundle, leftKeys[j]), TOL));
-        assertTrue(DoubleArrayMath.fuzzyEquals(boundInterp.parameterSensitivity(rightKeys[j]).toArray(),
-            baseInterp.getNodeSensitivitiesForValue(baseBundle, rightKeys[j]), TOL));
+        assertTrue(DoubleArrayMath.fuzzyEquals(boundInterp.parameterSensitivity(keys[j]).toArray(),
+            baseInterp.getNodeSensitivitiesForValue(baseBundle, keys[j]), TOL));
       }
     }
   }
 
   public void test_sameAsPrevious() {
     BoundCurveInterpolator bci =
-        CurveInterpolators.NATURAL_SPLINE.bind(X_DATA, Y_DATA, INT_EXTRAPOLATOR, INT_EXTRAPOLATOR);
+        CurveInterpolators.SQUARE_LINEAR.bind(X_DATA, Y_DATA, INT_EXTRAPOLATOR, INT_EXTRAPOLATOR);
     Extrapolator1D oldExtrap = Interpolator1DFactory.INTERPOLATOR_EXTRAPOLATOR_INSTANCE;
-    Interpolator1D oldInterp = new NaturalSplineInterpolator1D();
+    Interpolator1D oldInterp = new SquareLinearInterpolator1D();
     Interpolator1DDataBundle data = oldInterp.getDataBundle(X_DATA.toArray(), Y_DATA.toArray());
 
     for (int i = 0; i < 100; i++) {
-      final double x = RANDOM.nextDouble() * 20.0 - 10;
+      final double x = RANDOM.nextDouble() * 5.0;
```

Comment:
Doesn't this change defeat the point of the test? `RANDOM.nextDouble() * 20.0 - 10;` would have values outside the interpolation range, which is what is needed for this test.